lib/http/HttpClient_WinInet.cpp

```diff
@@ -136,6 +139,105 @@ class WinInetRequestWrapper
         return true;
     }
 
+    DWORD GetDWORDRegKey(HKEY hKey, const std::string& subKey, const std::string& value) {
+        DWORD data;
+        DWORD dataSize = sizeof(data);
+        LONG res = RegGetValueA(hKey, subKey.c_str(), value.c_str(), RRF_RT_REG_DWORD, NULL, &data, 
+            &dataSize);
+        if (res != ERROR_SUCCESS) {
+            return 0;
```

Comment:
How does a caller distinguish between a zero in the Registry value and an error?